solr/core/src/java/org/apache/solr/schema/DenseVectorField.java

```diff
@@ -218,22 +238,46 @@ public List<IndexableField> createFields(SchemaField field, Object value) {
 
   @Override
   public IndexableField createField(SchemaField field, Object vectorValue) {
+    FieldType denseVectorFieldType = getDenseVectorFieldType();
+
+
     if (vectorValue == null) return null;
     DenseVectorParser vectorBuilder = (DenseVectorParser) vectorValue;
     switch (vectorEncoding) {
       case BYTE:
         return new KnnByteVectorField(
-            field.getName(), vectorBuilder.getByteVector(), similarityFunction);
+            field.getName(), vectorBuilder.getByteVector(), denseVectorFieldType);
       case FLOAT32:
         return new KnnFloatVectorField(
-            field.getName(), vectorBuilder.getFloatVector(), similarityFunction);
+            field.getName(), vectorBuilder.getFloatVector(), denseVectorFieldType);
       default:
         throw new SolrException(
             SolrException.ErrorCode.SERVER_ERROR,
             "Unexpected state. Vector Encoding: " + vectorEncoding);
     }
   }
 
+  private FieldType getDenseVectorFieldType() {
```

Comment:
This turned out to be easier than I thought it would appear :-).

Please add a comment to reflect that this is a **hack** to work around Lucene's FloatVectorValues standard dimension check.  Maybe comment it might stop working in a future Lucene release.